DataFormats/TrackerCommon/src/LegacyTrackerTopology.cc

```diff
@@ -0,0 +1,233 @@
+#include "DataFormats/TrackerCommon/interface/LegacyTrackerTopology.h"
+
+#include "FWCore/Concurrency/interface/Xerces.h"
+#include <xercesc/parsers/XercesDOMParser.hpp>
+#include <xercesc/sax/HandlerBase.hpp>
+#include <xercesc/dom/DOM.hpp>
+#include "FWCore/Utilities/interface/Exception.h"
+
+using namespace xercesc;
+
+namespace {
+  // copy into a std::string and clean up the C-style string
+  inline std::string xmlc_to_stdstring(char* cstr)
+  {
+    std::string str{cstr};
+    XMLString::release(&cstr);
+    return str;
+  }
+  // transcode, convert into a std::string and clean up the intermediate C-style string
+  inline std::string xmlc_to_stdstring(const XMLCh* xcstr)
+  {
+    char* cstr = XMLString::transcode(xcstr);
+    std::string str{cstr};
+    XMLString::release(&cstr);
+    return str;
+  }
+
+  // RAII release the XMLCh*
```

Comment:
Some Xerces-C++ `char*`-equivalent (Unicode-aware, I think).
RAII is the technique that is also used in smart pointer classes like `std::unique_ptr`, lock handles etc.: make a helper class that acquires/releases a resource in the constructor and destructor, and make an instance as a local variable (the destructor will call the release when this variable goes out of scope, even in case an exception is thrown).
It prevents leaks, and it also avoids having to type a list of declarations at the top, and the corresponding list of cleanup statements at the bottom.